client/add/pokemon.ctrl.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Controller for handling the dialog to add a new box
+ * @return {function} A controller that contains 2 test elements
+ */
+module.exports = function ($mdDialog, Upload) {
+
+  this.visibilities = require('../../api/services/Constants.js').POKEMON_VISIBILITIES;
+  this.visibility = this.defaultPokemonVisibility;
+  this.box = this.defaultBox ? this.defaultBox.id : undefined;
+  this.file = {};
+
+  this.cancel = function() {
+    $mdDialog.cancel();
+  };
+
+  this.canAdd = function () {
+    return Boolean(this.visibility && this.file);
+  };
+
+  this.answer = function() {
+    this.file.upload = Upload.upload({
+      url: '/uploadpk6',
+      data: {visibility: this.visibility, pk6: this.file, box: this.box},
+    }).then(function (response) {
+      $mdDialog.hide(response.data);
+    }, function (response) {
+      console.log(response.status + ': ' + response.data);
+    }, function (evt) {
+      // Math.min is to fix IE which reports 200% sometimes
+      console.log(Math.min(100, parseInt(100.0 * evt.loaded / evt.total)));
```

Comment:
Yeah, I don't know why I have that there, it's probably just copied code from the example on the webpage. I'll remove it.